src/main/java/com/google/step/datamanager/UserManager.java

```diff
@@ -0,0 +1,51 @@
+package com.google.step.datamanager;
+
+import com.google.step.model.User;
+import java.util.Optional;
+
+/** A UserManager handling user datastore operations. */
+public interface UserManager {
+
+  /**
+   * Returns a User object when the user with the email logs in.
+   *
+   * @param email email of the user who logs in.
+   * @return User object with the email.
+   */
+  public User userLogin(String email);
```

Comment:
I think we should just have 2 separate methods `getUserByEmail` and `createUser`. And whoever is calling UserManager should first check if the user with the email using exists `getUserByEmail` and then create it if not.

Reply:
I was thinking that a user object should exist once it is logged in, so I wrote a separate method for login and user is read by id.